domain/src/main/kotlin/feature/ExportManager.kt

```diff
@@ -13,24 +13,36 @@ import kotlinx.coroutines.flow.first
 import kotlinx.coroutines.runBlocking
 import ltd.evilcorp.core.repository.MessageRepository
 import ltd.evilcorp.core.vo.Message
+import org.json.JSONArray
+import org.json.JSONObject
 
 class ExportManager @Inject constructor(
     private val messageRepository: MessageRepository,
 ) {
-    // TODO @Akito: Increment version programmatically on major changes.
-    private fun List<Message>.generateExportMessages(): ExportMessages {
-        return ExportMessages(
-            version = 1,
-            timestamp = SimpleDateFormat(
-                """yyyy-MM-dd'T'HH-mm-ss""",
-                Locale.getDefault(),
-            ).format(Date()),
-            entries = this,
+    private fun generateExportMessagesJString(messages: List<Message>): String {
+        val root = JSONObject()
+        root.put("version", 1)
+        root.put(
+            "timestamp",
+            SimpleDateFormat("""yyyy-MM-dd'T'HH-mm-ss""", Locale.getDefault()).format(Date()),
         )
-    }
 
-    private fun generateExportMessagesJString(messages: List<Message>): String =
-        jacksonObjectMapper().writerWithDefaultPrettyPrinter().writeValueAsString(messages.generateExportMessages())
+        val entries = JSONArray()
+        for (message in messages) {
+            val jsonMessage = JSONObject().apply {
+                put("publicKey", message.publicKey)
+                put("message", message.message)
+                put("sender", message.sender.toString())
+                put("type", message.type.toString())
+                put("correlationId", message.correlationId)
+                put("timestamp", message.timestamp)
+                put("id", message.id)
+            }
+            entries.put(jsonMessage)
+        }
+        root.put("entries", entries)
+        return root.toString(2)
+    }
```

Comment:
Code is maintainable, as long as the root object is this simple, so no problem.

Reply:
Yeah, there is also a kotlinx.serialization library I considered, but since this ended up being so simple even without any dependencies, I figured we could start like this and just bear it in mind in the future if things get more complex.